nixos/modules: services.syncthing add guiAddress parameter

mrVanDalo opened the pull request:
###### Motivation for this change

To make it possible to configure your syncthing from a remote machine.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
nixos/modules: services.syncthing add guiAddress parameter

Infinisil commented on nixos/modules/services/networking/syncthing.nix:
```diff
@@ -16,6 +16,14 @@ in {
       '';
 
+      guiAddress = mkOption {
+        type = types.string;
```
Use `types.str` instead

mrVanDalo replied:
I used types.string for consistency ( it is used in the other options as well )

Infinisil replied:
Change all of them to `str` then :)

mrVanDalo replied:
Done

Infinisil commented on nixos/modules/services/networking/syncthing.nix:
```diff
@@ -16,6 +16,14 @@ in {
+        type = types.string;
+        default = "127.0.0.1:8384";
+        description = ''
+          Override GUI listen address.
```
How about "Address to server the GUI on" instead? The fact that it overrides the default shouldn't be in the options description

mrVanDalo replied:
Done

peterhoeg reviewed (changes requested):
Looks good once @Infinisil 's concerns have been addressed.

Commit:
nixos/modules: services.syncthing use types.str instead of types.string

As Infinisil mentioned in https://github.com/NixOS/nixpkgs/pull/48119#discussion_r224974201

peterhoeg commented:
Looks good, thank you for your contribution @mrVanDalo

matthewbauer commented:
Could you add a release note that explains guiAddress change? This appears to override the default from configDir.

https://github.com/NixOS/nixpkgs/issues/49805

peterhoeg commented:
All, please note that the latest version .52 is just out which supports a "unix://" parameter to guiAddress - would be neat to support that here as part of this PR.
